 CRM/Core/BAO/Email.php | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CRM/Core/BAO/Email.php b/CRM/Core/BAO/Email.php
index 83c29be..f9d3326 100644
--- a/CRM/Core/BAO/Email.php
+++ b/CRM/Core/BAO/Email.php
@@ -68,6 +68,8 @@ class CRM_Core_BAO_Email extends CRM_Core_DAO_Email {
   static function add(&$params) {
     $email = new CRM_Core_DAO_Email();
     $email->copyValues($params);
+    
+    $is_primary = (is_numeric(CRM_Utils_Array::value('is_primary', $params)) ||empty($params['id']));
 
     // CRM-11006 move calls to pre hook from create function to add function
     if (!empty($params['id'])) {
@@ -96,6 +98,10 @@ WHERE  contact_id = {$params['contact_id']}
     self::holdEmail($email);
 
     $email->save();
+    
+    if($is_primary) {
+      CRM_Core_BAO_UFMatch::updateUFName($params['contact_id']);
+    }    
         
     // CRM-11006 move calls to pre hook from create function to add function
     if (!empty($params['id'])) {
